src/components/ApplicationGroup/Details/EnvCDDetails/EnvCDDetails.tsx

```diff
@@ -131,6 +132,30 @@ export default function EnvCDDetails({ filteredAppIds }: AppGroupDetailDefaultTy
             setHasMore(true)
             setHasMoreLoading(true)
         }
+        let _triggerId = deploymentHistoryResult.result?.cdWorkflows?.[0]?.id
+        const queryString = new URLSearchParams(location.search)
+        const deploymentStageType =
+            queryString.get('type') === STAGE_TYPE.PRECD ? DeploymentStageType.PRE : DeploymentStageType.POST
+
+        const triggeredHistoryResult = deploymentHistoryResult.result?.cdWorkflows?.filter((obj) => {
+            return obj.stage === deploymentStageType
+        })
+
+        if (triggeredHistoryResult?.[0]) {
+            _triggerId = triggeredHistoryResult[0].id
+        }
+
+        if (!triggerId && appId && pipelineId && deploymentHistoryResult?.result?.cdWorkflows?.length) {
+            replace(
```

Comment:
if we change path while rendering a component it will throw error in console

move this inside useEffect

Reply:
This is already nside useeffect.